open-policy-agent/authentication-user-management/shorten-tokens/template.yaml

```diff
@@ -0,0 +1,36 @@
+apiVersion: templates.gatekeeper.sh/v1beta1
+kind: ConstraintTemplate
+metadata:
+  name: k8sshortentoken
+  annotations:
+    description: Shorten the default Access Token Lifespan.
+spec:
+  crd:
+    spec:
+      names:
+        kind: K8sShortenToken
+  targets:
+    - target: admission.k8s.gatekeeper.sh
+      rego: |
+        package K8sShortenToken
+
+        violation[{"msg": msg}] {
+          input.review.kind.kind == "OAuth"
+          input.review.object.metadata.name == "cluster"
+          oauth := input.review.object
+          oauth.spec.tokenConfig.accessTokenMaxAgeSeconds > input.parameters.maxTokenTime
+          msg := sprintf("The OAuth CR token lifespace was set to %v, while it must be shorter than %v", [oauth.spec.tokenConfig.accessTokenMaxAgeSeconds, input.parameters.maxTokenTime])
+        }
+
+        violation[{"msg": msg}] {
+          input.review.kind.kind == "OAuth"
+          input.review.object.metadata.name == "cluster"
+          oauth := input.review.object
+          not oauth_token(oauth)
+          msg := sprintf("The OAuth CR token lifespace was set to 86400, while it must be shorter than %v, refer to 'https://docs.openshift.com/container-platform/4.6/authentication/configuring-internal-oauth.html' for more information", [input.parameters.maxTokenTime])
```

Comment:
@michaelkotelnikov 
Suggesting more accurate link that leads to token duration configuration specifically: https://docs.openshift.com/container-platform/4.6/authentication/configuring-internal-oauth.html#oauth-configuring-internal-oauth_configuring-internal-oauth

Reply:
@igorkchyts Added.